src/main/scala/com/typesafe/scalalogging/Logger.scala

```diff
@@ -126,6 +128,8 @@ final class Logger private (val underlying: Underlying) extends Serializable {
 
   def warn(marker: Marker, message: String, args: Any*): Unit = macro LoggerMacro.warnMessageArgsMarker
 
+  def whenWarn(body: Unit): Unit = macro LoggerMacro.warnCode
```

Comment:
let's name this whenWarnEnabled

Reply:
OK, I renamed all functions from when[level] to when[level]Enabled